packages/trader/src/TraderConfig.ts

```diff
@@ -2,15 +2,40 @@ import Web3 from "web3";
 import assert from "assert";
 
 export interface ProcessEnv {
-  [key: string]: string | undefined;
+  [key: string]: any;
```

Comment:
I got this error when it was set to `string | undefined`
<img width="1222" alt="image" src="https://user-images.githubusercontent.com/12886084/109270462-ba96f100-7816-11eb-9a8b-1e2b08cf963c.png">

Reply:
oh i see yea, you probably can get around this by either throwing an error if `DS_PROXY_FACTORY_ADDRESS` is undefined, or setting a default string like calling with `DS_PROXY_FACTORY_ADDRESS || ''`

one thing that is really annoying about typescript is i dont think it will detect `assert` as properly gaurding against null checks. You may have to explicitly use `if(DS_PROXY_FACTORY_ADDRESS == null) throw new Error('blah')`

Reply:
^+1. This is just typescript forcing you to "narrow" the type (aka create a logic branch for string and for undefined) before you use it where a string is expected.